docker/deploy.sh

```diff
@@ -1,56 +1,150 @@
+    then
+        echo " "
+        echo "Press Ctrl+C to quit out of the log view."
+        sleep 2
+        docker service logs --raw -f op-scim_scim 2>/dev/null
+    else
+        echo "Skipping logs..."
+        echo "You can view the logs manually by running: docker service logs --raw -f op-scim_scim"
+    fi
 }
 
-echo 'Please ensure you have read the README before continuing.'
-echo 'Which docker deployment method are you using?'
-read -p '[compose] or [swarm]: ' docker_path
-
-if [ "$docker_path" == "compose" ];
-then 
+# Begin main script
+
+docker_path=$(dirname $(realpath $0))
+
+echo "Initiating 1Password SCIM Bridge Deployment to Docker Swarm"
+echo " "
+echo "Please specify the following options."
+
+while :
+do
+    read -p "Docker Swarm or Docker Compose? [swarm/compose]: " docker_type
+    if [[ "$docker_type" =~ ^(swarm|compose)$ ]]; then
+        break
+    fi
+    echo "$docker_type is not a valid input. Please select either 'swarm' or 'compose'."
+done
+
+
+while :
+do
+    read -p "Fully-qualified domain name (FQDN) you are deploying to [e.g: 'op-scim.example.com']: " domain_name
+    if [[ $domain_name = *.* ]]; then
+        break
+    fi
+    echo "Please enter a fully-qualified domain name."
+done
+
+while :
+do
+    read -p "Path to your scimsession file: " scimsession_file
+    if [[ -f "$scimsession_file" ]]
+    then
+        break
+    fi
+    echo "File '$scimsession_file' does not exist at that path, please try again." >&2
+done
+
+echo " "
+echo "Using the following parameters to deploy the SCIM Bridge"
+echo "Deployment type:" $docker_type
+echo "scimsession file path:" $scimsession_file
+echo "Domain name:" $domain_name
+
+while ! [[ "$proceed" =~ ^([yY][eE][sS]|[yY])$ || "$proceed" =~ ^([nN][oO][nN]) ]]; do
+    read -p "Does this look correct? [Y/n]: " proceed
+    if [[ "$proceed" =~ ^([nN][oO][nN])$ ]]
+    then
+        echo "Exiting..."
+        exit 0 
+    fi
+done
+
+# place the domain name into the deployment file, in a backup
+docker_file_path=$docker_path/$docker_type
+docker_original_file=$docker_file_path/docker-compose.yml
+docker_deploy_file=$docker_file_path/docker-compose.yml.deploy
+sed s/{YOUR-DOMAIN-HERE}/$domain_name/g $docker_original_file > $docker_deploy_file
+
+# run the function associated with the Docker type selected
+if [[ "$docker_type" == "compose" ]]
+then
     run_docker_compose
-elif [ "$docker_path" == "swarm" ]
+elif [[ "$docker_type" == "swarm" ]]
 then
     run_docker_swarm
-else 
-    echo "Invalid docker manager. Please use either docker-compose or docker-swarm."
-    exit
 fi
+
+echo " "
+echo "Deployment of the 1Password SCIM Bridge is complete!"
+echo " "
+echo "If you have any issues deploying the SCIM Bridge, please either reach out to 1Password Business Support, or look through our helpful discussion forums: https://discussions.agilebits.com/categories/scim-bridge"
+echo " "
+echo "Back up of deployment docker-compose.yml file saved to" $docker_deploy_file
```

Comment:
If you are `sed s/{YOUR-DOMAIN-HERE}/$domain_name/g $docker_original_file > $docker_deploy_file` is the `$docker_deploy_file` really a backup? Do you mean the `#docker_original_file`?

Reply:
The idea here is that I'm point them towards the file that was actually used for their deployment, which is `$docker_deploy_file`. 

I'm realizing now that this might not be clear between the filenames the documentation use vs what's used in the script. I'll change that up.